src/app/access-requests/features/access-request-field-edit/access-request-field-edit.component.ts

```diff
@@ -71,24 +82,33 @@ export class AccessRequestFieldEditComponent implements OnInit {
     this.field.update(() => this.#defaultValue());
   }
 
-  #getValue = () => {
-    if (
-      this.name() === 'ticket_id' &&
-      this.field() &&
-      (this.field() as string).startsWith(this.#baseTicketUrl)
-    ) {
-      this.field.set((this.field() as string).slice(this.#baseTicketUrl.length));
-    }
-    return this.field();
-  };
-
   edit = () => {
     this.isOpen.set(true);
   };
 
-  changed = () => {
+  /**
+   * Remove any URL prefix from the field value if this is a ticket ID
+   * and set the proper error message if the field is invalid.
+   * @param control - the form control for the field
+   */
+  #handleControl = (control: FormControl) => {
+    if (this.name() === 'ticket_id') {
+      // if the field is prefixed with (parts of) the base ticket URL, remove that prefix
+      const baseUrl = this.#baseTicketUrl;
+      let value = this.field() || '';
+      const i = value.lastIndexOf('/');
+      if (baseUrl && i >= 0 && baseUrl.endsWith(value.substring(0, i + 1))) {
+        value = value.substring(i + 1);
+        control.setValue(value);
+      }
+      this.validationError.set(control.invalid ? ERROR_TICKET_ID : '');
+    }
+  };
+
+  changed = (control?: FormControl) => {
```

Comment:
By the way, and this is true also for the date editing, if you click cancel after you have already saved a previous edit, the data goes back to what was originally there before the first edit, rather than the value saved after the first edit.

Reply:
This might have to do with the way mocking works, but I have not yet explored it thoroughly.